Spatial.gd

```diff
@@ -816,77 +826,58 @@ func _on_NewPathPoint_pressed():
 		var z_accurate_player_position = player_position
 		z_accurate_player_position.z = -z_accurate_player_position.z
 		self.currently_active_path.add_point(z_accurate_player_position)
-		self.currently_active_path_2d.add_point(Vector2(self.player_position.x, -self.player_position.z))
+		self.currently_active_path2d.add_point(z_accurate_player_position)
 
 func _on_NodeEditorDialog_hide():
-	self.currently_selected_node = null
+	self.currently_selected_gizmo = null
 	clear_adjustment_nodes()
 	_on_Dialog_hide()
 
 
 func _on_DeleteNode_pressed():
-	if self.currently_selected_node.point_type == "icon":
-		self.currently_selected_node.object_link.get_parent().remove_child(self.currently_selected_node.object_link)
-		self.currently_selected_node.object_link.queue_free()
-	elif self.currently_selected_node.point_type == "path":
-		var path =   self.currently_selected_node.object_link
-		var path2d = self.currently_selected_node.object_2d_link
-		var index =  self.currently_selected_node.object_index
-
-		path.remove_point(index)
-		path2d.remove_point(index)
+	self.currently_selected_node3d.remove(self.currently_selected_index)
```

Comment:
I feel like both currently_selected_node3d and currently_selected_gizmo are allowed to be null. Should we have null checks for them?

Reply:
It "should" never be null when the button is clicked because the button "should" be deactivated if there is no selected gizmo.

But the variable probably "can" be null

Reply:
True, thinking through it, I considered what would happen if the gizmo was null and it turns out, nothing here because the deselected function doesn't use the gizmo at all. It just resets all the defaults. Therefor I think it should be changed. I will add checks for the places where the gizmo is actually used.